Formula/lib/libtomcrypt.rb

```diff
@@ -19,8 +19,12 @@ class Libtomcrypt < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "8957df3d2611c3979bf19a68c0b0f1cab117c3a434f979b0a6a5cf21f4bc560e"
   end
 
+  depends_on "gmp"
+  depends_on "libtommath"
+
   def install
-    system "make", "test"
+    system "make", "CFLAGS=-DUSE_LTM -DLTM_DESC -DGMP_DESC"
+    system "make", "test", "CFLAGS=-DUSE_LTM -DLTM_DESC -DGMP_DESC", "EXTRALIBS=-ltommath"
     system "make", "install", "PREFIX=#{prefix}"
     pkgshare.install "test"
     (pkgshare/"tests").install "tests/test.key"
```

Comment:
Did you forget to add `-lgmp`? Also, let's build shared libraries too. That will also need `depends_on "libtool" => :build`.

```suggestion
    ENV.append_to_cflags "-DUSE_LTM -DLTM_DESC -DGMP_DESC"
    ENV.append "EXTRALIBS", "-lgmp -ltommath"
    system "make", "-f", "makefile.shared"
    system "make", "-f", "makefile.shared", "test"
    system "./test"
    system "make", "-f", "makefile.shared", "install", "PREFIX=#{prefix}"
```

Reply:
See also: https://github.com/Homebrew/homebrew-core/pull/22218.

Reply:
Hi @ZhongRuoyu 

Thanks a lot for your help but I am not sure to understand all your proposed changes. I am not familiar enough with all the Ruby code in formulae.

The original formula was not written by me. I just wanted to reenable asymmetric crypto in libtomcrypt (without it, it broke the port of some test applications). I didn't want to interfere with the structure of the existing formula and I tried to find the minimal required changes.

I have briefly read the discussion in #22218. It is weird that the build with libtommath and gmp was made optional and finally completely removed. This makes libtomcrypt useless in asymmetric crypto. It needs at least one MP math backend.

I also read the discussion whether gmp should be used or not. I think that libtomcrypt should be built with the two. This is the choice of the application to select the proper MP backend at run time. Traditionally, libtomcrypt was used with libtommath (same author) but, nowadays, gmp is much faster than libtommath, twice as fast in average according to my recent tests. This is why the two must be referenced during the build.

Concerning your proposed changes, you run the test application as part of the build. Why not, but would this fix the issue in the test part of the formula? I assume that this test part is the one that is run by `brew test`, which would still fail in the CI then, isn't it?

Reply:
> I just wanted to reenable asymmetric crypto in libtomcrypt (without it, it broke the port of some test applications). I didn't want to interfere with the structure of the existing formula and I tried to find the minimal required changes.

Yes, that's understandable. What I am proposing, on top of your changes, are: 1. to add shared libraries because [it is Homebrew's policy](https://docs.brew.sh/Acceptable-Formulae#static-libraries) that shared libraries should be preferred; 2. to run tests during the build because that works around the "Library not loaded" issue, and more importantly, verify that the crypto library behaves as expected.

> It is weird that the build with libtommath and gmp was made optional and finally completely removed.

> I also read the discussion whether gmp should be used or not.

Yes, that's because `homebrew/core` no longer allows optional dependencies. We can re-enable them like what you're proposing here, though.

> but would this fix the issue in the test part of the formula?

Yes, it would. The direct reason why you encountered the "Library not loaded" failure was the moving of the test executable, from `pkgshare` to the test path. (Your changes were not the root cause.)

Reply:
> However, a consequence of 2. is that we need a new test that builds a simple program against the built library. Are you willing to do that? If not, I can come up with one.

I will do it. I have plenty of libtomcrypt apps and I can extract a small one.

Reply:
Thanks! Sorry for the language; `s/Are you willing to/Would you like to/`.

Reply:
> Sorry for the language

Don't worry, I am not a native English either.
I have pushed an update. It builds and passes the test locally. Let's see the CI...

Reply:
All CI checks passed this time. Thanks @ZhongRuoyu for your help. The PR just needs your approval now.